src/main/java/com/SAS/League/Season.java

```diff
@@ -0,0 +1,237 @@
+     */
+    public LeagueRankPolicy getRankPolicy(League league) {
+        return this.rankPolicy.get(league);
+    }
+
+    /**
+     * @param league the league that the games will be take place in
+     * @param games  the list of the games
+     */
+    public void addGamesList(League league, ArrayList<Game> games) {
+        if (this.gamesList.get(league) == null) {
+            boolean wasAdded = false;
+            GamesList game = new GamesList(league, this, games);
+            this.gamesList.put(league, game);
+            for (int i = 0; i < this.leaguesList.size(); i++) {
+                if (this.leaguesList.get(i).getName().equals(league)) {
+                    this.leaguesList.get(i).addGamesList(this, games);
+                    wasAdded = true;
+                }
+            }
+            if (!wasAdded) {
+                this.gamesList.remove(game); //if the league doesn't exist
+            }
+        }
+    }
+
+
+    /**
+     * @param team   which team the budgets belong to
+     * @param budget the budget of the input team for this season
+     */
+    public void addBudget(Team team, int budget) {
+        if (this.budgets.get(team) == null) {
+            boolean wasAdded = false;
+            Budget budgetToAdd = new Budget(team, this, budget);
+            this.budgets.put(team, budgetToAdd);
+            for (int i = 0; i < this.teamsList.size(); i++) {
+                if (this.teamsList.get(i).getName.equals(team)) {
+                    this.teamsList.get(i).addBudget(this.year, budget);
+                    wasAdded = true;
+                }
+            }
+            if (!wasAdded) {
+                this.budgets.remove(budgetToAdd); //if the team doesn't exist
+            }
+        }
+    }
+
+    /**
+     * @param league the league that the table is relevant to
+     * @param table: the league table for this the league
+     */
+    public void addTable(League league, Table table) {
+        if (this.tables.get(league) == null) {
+            boolean wasAdded = false;
+            this.tables.put(league, table);
+            for (int i = 0; i < this.leaguesList.size(); i++) {
+                if (this.leaguesList.get(i).getName().equals(league)) {
+                    this.leaguesList.get(i).addTable(this, table);
+                    wasAdded = true;
+                }
+            }
+            if (!wasAdded) {
+                this.tables.remove(table); //if the league doesn't exist
+            }
+        }
+    }
+
+    /**
+     * @param league      the league that the games policy is relevant to
+     * @param gamesPolicy the game policy you want to add
+     */
+    public void addGamePolicy(League league, GamesPolicy gamesPolicy) {
+        if (this.gamesPolicy.get(league) == null) {
+            boolean wasAdded = false;
+            this.gamesPolicy.put(league, gamesPolicy);
+            for (int i = 0; i < this.leaguesList.size(); i++) {
```

Comment:
I think instead of the for loop you can just call to leagu.addPolicy(this, gamesPolicy);
you don't to search it because you already have it.
And it can be changed in all the functions in League & Season of adding policies, games and tables.